server/src/main/java/org/elasticsearch/indices/recovery/PeerRecoverySourceService.java

```diff
@@ -74,6 +77,19 @@ public PeerRecoverySourceService(TransportService transportService, IndicesServi
             new StartRecoveryTransportRequestHandler());
     }
 
+    @Override
+    protected void doStart() {
+    }
+
+    @Override
+    protected void doStop() {
+        ongoingRecoveries.awaitEmpty();
```

Comment:
Does this awaitEmpty block a node from closing? Should we just cancel all ongoing recoveries?

Reply:
@dnhatn it was my understanding that we already do cancel all ongoing recoveries when shutting down the indices service (that's what happens in practice at the moment it seems) => I didn't think I need to take any action here.

> Does this awaitEmpty block a node from closing? 

Yes, but see above. I think this is fine. In the end, if the recoveries aren't properly stopped we will eventually interrupt this wait in the test cluster code and throw an exception so this won't silently cause trouble in the future either.